src/claimtrie.cpp

```diff
@@ -2,10 +2,16 @@
 #include "coins.h"
 #include "hash.h"
 
-#include <boost/scoped_ptr.hpp>
 #include <iostream>
 #include <algorithm>
 
+#define HASH_BLOCK 'h'
+#define CURRENT_HEIGHT 't'
+
+#include "claimtriedb.cpp" /// avoid linker problems
```

Comment:
No, this is broken.

Reply:
I agree that this is a bad pattern. This should technically cause linker problems due to double definition.

Reply:
It looks like but is not. Template has a external linking and it has few *hacks* to prevent problems:
1. Write template in class body - this is ugly and is not a proper C++ fix
2. Write a 3th file only with template implementation then include only. This adds an inconsistency between .h and .cpp with added .impl
3. Mark normal functions as inline *do* the trick and it's the natural way to keep consistency of .h and .cpp files.

Reply:
Including a cpp file from another means incorrect/bad organization.  That is what I meant by broken.